src/OneSignal.ts

```diff
@@ -731,11 +732,32 @@ export default class OneSignal {
       throw new InvalidArgumentError('url', InvalidArgumentReason.Malformed);
     if (!ValidatorUtils.isValidUrl(icon, { allowEmpty: true, requireHttps: true }))
       throw new InvalidArgumentError('icon', InvalidArgumentReason.Malformed);
+    if (!icon) {
+      // get default icon
+      const icons = await MainHelper.getNotificationIcons();
+      icon = getPlatformNotificationIcon(icons);
+    }
 
-    if (subscription.deviceId) {
-      await OneSignalApi.sendNotification(appConfig.appId, [subscription.deviceId], { en : title }, { en : message },
-                                               url, icon, data, buttons);
+    data = {
+      ...data,
```

Comment:
We needed actually to nest the data in the data object, as we want the click event to produce the format in the docs:
https://documentation.onesignal.com/v9.0/docs/web-push-sdk#addlistenerfornotificationopened-event

However this is what I am seeing instead for the click event when I used `OneSignal.sendSelfNotification("title", "msg", "https://localhost:4001/?app_id=a3268fe7-cc94-4f25-ab09-867b957d7fe2", "https://localhost:4001/icon", {key1: "value1"})` to create it.
```js
{
    "key1": "value1",
    "url": "https://localhost:4001/?app_id=a3268fe7-cc94-4f25-ab09-867b957d7fe2"
}
```